src/mtcp/mtcp_restart.c

```diff
@@ -495,13 +496,90 @@ mysetauxval(char **evp, unsigned long int type, unsigned long int val)
   return -1;
 }
 
-#define PAGESIZE 4096 /* We hardwire this, since we can't make libc calls. */
-// We choose this address based on our observation on CORI that it won't overlap
-// with any other memory segments. We might change this address in the future,
-// for now, it doesn't matters.
-#define vvarStartTmp ((0x40000) - 5*PAGESIZE)
-#define vdsoStartTmp ((0x40000) - 2*PAGESIZE)
+/* We hardwire these, since we can't make libc calls. */
+#define PAGESIZE 4096
+#define ROUNDADDRUP(addr, size) ((addr + size - 1) & ~(size - 1))
+// I think we're forced to use global variables, since
```

Comment:
Could you elaborate that why the global variable would persist if the `unmap_memory_areas_and_restore_vdso` function is unmapping all mtcp regions including heap and data sections?

Reply:
i'm not too sure what you mean by this. i'm using global variables here because another function uses it later on.

Reply:
It is a bit informal to say "I think we're forced". I meant to update the comment stating the reason instead of a dubious statement. Rest LGTM :)

Reply:
ah ok, very true.